### ISSUE TITLE
Detect overly complex relative require

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#516]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] ChefDeprecations - A cop that alerts on a deprecation in the Chef Infra Client
- [ ] ChefCorrectness - A cop that alerts when incorrect coding behavior is being used
- [ ] ChefSharing - A cop that alerts to missing functionality necessary for sharing cookbooks with others
- [ ] ChefStyle - A cop that alerts to a style best practice
- [ ] ChefModernize - A cop that alerts when a cookbook can be simplified or modernized with new functionality
- [ ] ChefEffortless - A cop that alerts on code that must be resolved to move to the Effortless pattern

### Describe the new cop:
require_relative can be used to simplify relative requires and they're fast

### What it would trigger on:

```ruby
require File.expand_path("../azure_base", __FILE__)
```

### What it would autocorrect to if applicable

```ruby
require_relative "azure_base"
```

**Comment by tyler-ball** [comment]
@tas50 In the autocorrect example that should be `require_relative "../azure_base"` to match the previous example, correct?

**Comment by lamont-granquist** [comment]
since it is relative to __FILE__ i'm pretty sure the ".." eats only the filename and so "azure_base" would be in the same directory ==  `#{File.dirname(__FILE__)}/azure_base`

**Comment by tas50** [comment]
Fixed in #535
